worker/upgradedatabase/worker.go

```diff
@@ -213,22 +212,26 @@ func (w *upgradeDB) upgradeDone() bool {
 	return false
 }
 
-func (w *upgradeDB) runUpgrade() {
+func (w *upgradeDB) runUpgrade() error {
```

Comment:
Have we tested the behaviour on the primary when the upgrade is running and it switches to secondary?
I wonder if we can introduce a wrench to pause an upgrade step for say 1 minute, enough time to manually kill the primary, and observe the behaviour, and then write a CI test specifically to test this for real, ie have the CI test add the wrench and check the upgrade survives

Reply:
I have tested this manually and am just about to add QA steps.